### PR TITLE
Handle masks in VectorAPIExpansion

### PR DESCRIPTION
**Pull request by gita-omr**
- only transform intrinsics that have NULL masks

Signed-off-by: Gita Koblents <koblents@ca.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
jenkins test sanity plinux,alinux64 jdk17

**Comment by gita-omr** [comment]
Added signature according to the guidelines.

**Comment by gita-omr** [comment]
jenkins test sanity plinux,alinux64 jdk18

**Review by knn-k — APPROVED** [review]

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp [hunk 1]
```diff
@@ -1666,12 +1674,12 @@ TR_VectorAPIExpansion::methodTable[] =
...
    {broadcastCoercedIntrinsicHandler, TR::NoType, Vector, {Unknown, elementType, numLanes, Unknown, Unknown, Unknown}},  // jdk_internal_vm_vector_VectorSupport_broadcastCoerced
-   {compareIntrinsicHandler, TR::NoType, Vector, {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector, Unknown}}, // jdk_internal_vm_vector_VectorSupport_compare
-   {ternaryIntrinsicHandler, TR::NoType, Vector,  {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector}},  // jdk_internal_vm_vector_VectorSupport_ternaryOp
-   {unaryIntrinsicHandler, TR::NoType, Vector,  {Unknown, Unknown, Unknown, elementType, numLanes, Vector}},  // jdk_internal_vm_vector_VectorSupport_unaryOp
+   {compareIntrinsicHandler, TR::NoType, Vector, {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector, Mask}}, // jdk_internal_vm_vector_VectorSupport_compare
+   {ternaryIntrinsicHandler, TR::NoType, Vector,  {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector, Mask}},  // jdk_internal_vm_vector_VectorSupport_ternaryOp
```
Comment (Akira1Saitoh) [inline runtime/compiler/optimizer/VectorAPIExpansion.cpp:1681]: Hi, @gita-omr 
Shouldn't this line have 3 Vectors arguments as below? I noticed FMA was not vectorized as-is, but if I change this line, it was vectorized.

```
{ternaryIntrinsicHandler, TR::NoType, Vector,  {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector, Vector, Mask}},
```

Comment (Akira1Saitoh) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp:1681]: Opened https://github.com/eclipse-openj9/openj9/pull/14962.

Comment (gita-omr) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp:1681]: > Hi, @gita-omr Shouldn't this line have 3 Vectors arguments as below? I noticed FMA was not vectorized as-is, but if I change this line, it was vectorized.
> 
> ```
> {ternaryIntrinsicHandler, TR::NoType, Vector,  {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector, Vector, Mask}},
> ```

@Akira1Saitoh , you are right. It should be three vector arguments.

**Pull request merged**